ui/plugin-lifecycle/src/main/services/organisation.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from "@angular/core";
+import { Http, Response, Headers, RequestOptions } from "@angular/http";
+import { Organisation } from "../interfaces/Organisation";
+import { Subject, Observable } from "rxjs";
+import { AuthService } from "./auth.service";
+import { XMLHelper } from "../classes/XMLHelper";
+
+@Injectable()
+export class OrganisationService {
+    private _orgs: Organisation[] = [];
+    private _orgsSubject = new Subject<Organisation[]>();
+
+    constructor(
+        private http: Http,
+        private authService: AuthService
+    ) {
+        this.authService.auth()
+            .then(() => {
+                this.loadOrgs();
+            })
+            .catch((err) => {
+                console.error(err);
+            });
+    }
+
+    get orgs(): Organisation[] {
+        return this._orgs;
+    }
+
+    set orgs(orgs: Organisation[]) {
+        this._orgs = orgs;
+    }
+
+    private addOrg(org: Organisation) {
+        Object.keys(org).forEach((key) => {
+            try {
+                org[key] = JSON.parse(org[key])
+            } catch (error) {
+                org[key] = org[key]
+            }
+        });
+
+        this.orgs.push(org);
+        this._orgsSubject.next(this.orgs);
+    }
+
+    public watchOrgs(): Observable<Organisation[]> {
+        return this._orgsSubject.asObservable();
+    }
+
+    public refresh() {
+        this.loadOrgs();
+    }
+
+    private loadOrgs(): void {
+        this.getAllOrganisations("https://bos1-vcd-sp-static-200-117.eng.vmware.com", this.authService.getAuthToken())
```

Comment:
Another place where you'll need to rely on the injected container hooks.  See the latest seed project for example usage.

Reply:
This is already replaced but in plugin-lifecycle branch.